numpy/lib/function_base.py

```diff
@@ -514,7 +514,9 @@ def average(a, axis=None, weights=None, returned=False):
         avg = a.mean(axis)
         scl = avg.dtype.type(a.size/avg.size)
     else:
-        a = a + 0.0
+        if np.issubclass_(a.dtype.type, np.inexact) is False:
```

Comment:
Will do, is there a technical difference between <code>X is False</code> and <code>not X</code> that I am not aware of?